src/Libraries/Microsoft.Extensions.AI.Evaluation.Safety/ContentSafetyService.cs

```diff
@@ -20,15 +20,10 @@ internal sealed partial class ContentSafetyService(ContentSafetyServiceConfigura
     private const string APIVersionForServiceDiscoveryInHubBasedProjects = "?api-version=2023-08-01-preview";
     private const string APIVersionForNonHubBasedProjects = "?api-version=2025-05-15-preview";
 
-    private static HttpClient? _sharedHttpClient;
-    private static HttpClient SharedHttpClient
-    {
-        get
-        {
-            _sharedHttpClient ??= new HttpClient();
-            return _sharedHttpClient;
-        }
-    }
+    private static HttpClient SharedHttpClient =>
+        field ??
+        Interlocked.CompareExchange(ref field, new(), null) ??
+        field;
```

Comment:
Since `CompareExchange` always returns the original value (which would either be `null` or an existing valid `HttpClient`) can we simplify to use a single `??` check?

```suggestion
    private static HttpClient SharedHttpClient =>
        Interlocked.CompareExchange(ref field, new(), null) ??
        field;
```

Reply:
(The pattern with two `??` is interesting :) I don't think I have come across this before - I wish there were an easier way to express this - or if there were a `Interlocked.CompareExchangeAndReturnTheNewValueInsteadOfTheOriginalOne` 😄)